modules/op-constructing-pipelines-using-existing-tasks.adoc

```diff
@@ -0,0 +1,43 @@
+* The Parameters for a Task are prepopulated based on the specifications of the Task. If required, reference the additional Parameters added to the Pipeline in the Task.
+
+.Procedure
+
+. In the *Add* view of the *Developer* perspective, click the *Pipeline* tile to see the *Pipeline Builder* page.
+. Enter a unique name for the pipeline.
+. Select a Task from the *Select task* list to add a Task to the pipeline. For this example, select a *s2i-nodejs* Task.
+.. To add sequential Tasks to the pipeline, click the + icon to the right or left of the Task, and from the *Select task* list, select the Task you want to add to the pipeline. For this example, use the + icon to the right of the *s2i-nodejs* Task to add an *openshift-client* Task.
+.. To add a parallel Task to the existing Task, click the + icon displayed below the Task, and from the *Select Task* list, select the parallel Task you want to add to the pipeline.
++
+image::pipeline-builder.png[Pipeline Builder]
++
+. Click *Add Resources* to specify the name and type of Resources that the Pipeline will need to use during execution. These resources are then used by the Tasks in the Pipeline as Inputs and Outputs.
+For this example:
+.. Add an Input resource. In the *Name* field, enter `Source`, and from the *Resource Type* drop-down list, select *Git*.
+.. Add an Output resource. In the *Name* field, enter `Img`, and from the *Resource Type* drop-down list, select *Image*.
+. The *Parameters* for a Task are prepopulated based on the specifications of the Task. If required, use the *Add Parameters* link to add additional parameters .
++
+. A *Missing Resources* warning is displayed on a Task if the resources for the Task are not specified. Click the *s2i-nodejs* Task to see the side panel with details for the Task.
++
+image::op-pipeline-builder-task-details.png[Tasks details in Pipelines Builder]
++
+. In the Task side panel, specify the resources and parameters for it:
+
+.. In the *Input Resources* -> *Source* section, click the *Select Resources* list to see the resources you added to the pipeline displayed as options, and select *Source*.
+.. In the *Output Resources* -> *Image* section, click the *Select Resources* list, and select *Img*.
+.. If required, in the *Parameters* section, add more parameters to the default ones, by using the `$(params.<param-name>)` syntax.
+. Similarly, add an Input resource for the *openshift-client* Task.
+. Click *Create* to create the Pipeline. You are redirected to the *Pipeline Details* page, displaying the details of the created pipeline. You can now use the *Action* button to start the Pipeline.
+
+Optionally, you can also use the *Edit YAML* link, on the upper right of the *Pipeline Builder* page, to directly modify the Pipeline YAML file in the console. You can also use the operator-installed, reusable snippets and samples to craft better Pipelines.
```

Comment:
```suggestion
Optionally, you can also use the *Edit YAML* link, on the upper right of the *Pipeline Builder* page, to directly modify a Pipeline YAML file in the console. You can also use the operator-installed, reusable snippets and samples to craft better Pipelines.
```
Small but important nit. It's not _the_ Pipeline YAML it's just a template we have... first pass (new creations) is the template and on edit Pipeline it's just back to the YAML that is tied to the Pipeline resource you're editing.